### ISSUE TITLE
pil_to_tensor() raises a "NumPy array is not writeable" warning 

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#4508]
### 🐛 Describe the bug

Steps to reproduce:

```python
import torchvision
from PIL import Image
img = Image.open("test/assets/encode_jpeg/grace_hopper_517x606.jpg")
torchvision.transforms.functional.pil_to_tensor(img)
```

Raises:
```
./vision/torchvision/transforms/functional.py:177: UserWarning: The given NumPy array is not writeable, and PyTorch does not support non-writeable tensors. This means you can write to the underlying (supposedly non-writeable) NumPy array using the tensor. You may want to copy the array to protect its data or make it writeable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at  ./torch/csrc/utils/tensor_numpy.cpp:187.)
  img = torch.as_tensor(np.asarray(pic))
```

https://github.com/pytorch/vision/blob/f483e71b512e0f9dd5d6eb531fb602581ffb39a6/torchvision/transforms/functional.py#L177 

There are a few options including copying the data, suppressing the warnings etc (see discussion below). The person who picks this up can comment on this issue which solution they are going for.

### Versions

Latest main (f749835057ec16886f16d5533928e269860f1760)

cc @vfdev-5 @datumbox

**Comment by NicolasHug** [comment]
Do we know what caused this? Was there a change made to PIL recently?

**Comment by datumbox** [comment]
Could also be from PyTorch. It has happened before: https://github.com/pytorch/pytorch/issues/47160

**Comment by fmassa** [comment]
This is a change from PyTorch. This will not be a problem for us except if we perform any in-place operations in the returned tensor (which I think almost never happens). So we could probably just silence this warning if possible.

**Comment by datumbox** [comment]
Sounds reasonable. I've updated the ticket to indicate that.

**Comment by NicolasHug** [comment]
Do you know which PR / diff caused the change @fmassa ?

I agree it shouldn't be a problem in most cases. That being said, if we almost never make inplace operations that means we almost always make a copy, so we might as well use @datumbox earlier suggestion to use `np.array` instead?

**Comment by fmassa** [comment]
It was added in https://github.com/pytorch/pytorch/pull/33615

Another option is to do something like

```python
nparray.setflags(write=False)
```
but we should ensure that the user doesn't perform any in-place operations, otherwise the underlying PIL image will be modified.

We have the choice here between efficiency (no copy) and being on the safe side (with copy).

I'm ok going for the copy, but we need to keep in mind that we need to make the transforms faster in the future (as they might already be a bottleneck in newer GPUs)

**Comment by NicolasHug** [comment]
Alternatively we could implement both with a new `force_copy` flag passed to `pil_to_tensor` or something like that.
No strong opinion on my side, since apparently the warning has been there for a while.

**Comment by jdsgomes** [comment]
I am inclined to add a `force_copy` flag but keep the warning in case the flag is `False`, so it is clear to the caller that the underlying array can be modified. Any thoughts?

**Comment by EricPHassey** [comment]
Oddly enough I'm still getting the warning message when calling this. What version was this added to?

**Comment by jdsgomes** [comment]
> Oddly enough I'm still getting the warning message when calling this. What version was this added to?

It should be included from v0.11.0 (https://github.com/pytorch/vision/commits/v0.11.0)
